Objects/tupleobject.c

```diff
@@ -76,15 +76,11 @@ _PyTuple_DebugMallocStats(FILE *out)
 #endif
 }
 
-PyObject *
-PyTuple_New(Py_ssize_t size)
+static PyObject *
+tuple_new_internal(Py_ssize_t size)
```

Comment:
The current implementation is dangerous: it tracks in the GC a tuple with uninitialized items.

You should either move _PyObject_GC_TRACK() call into ther caller or initialize size to 0 and set the size once items are initialized.

I propose to rename tuple_new_internal() to tuple_new_untracked() as an hint that this function is special, and move _PyObject_GC_TRACK() call to the caller.